language/Pattern-Matching.md

```diff
@@ -176,6 +176,36 @@ compare _ _ = EQ
 
 (The name `otherwise` is a synonym for `true` commonly used in guards.)
 
+Guard Limitations in `let`
+----------------------------
+
+Guards are incompatable with Constructor Patterns in `let`. For example, the following function using a `Tuple` constructor pattern will not compile:
+```purs
+-- This doesn't work
+f1 :: Int
+f1 =
+  let
+    (Tuple a b)
+      | false = Tuple 1 2
+      | otherwise = Tuple 3 4
+  in
+    a
+```
```

Comment:
Some of the capitalization is a bit odd -- we don't usually capitalize "guard" or "constructor". What about:

> Constructor patterns with guards are not supported in `let` expressions. For example, this does not compile:

We could also trim this example a little -- the example from https://github.com/purescript/purescript/pull/3200 is pretty good:

```purs
f :: Int -> Int -> Int
f a b =
  let 
    Tuple x y
      | a < b = Tuple a b
      | otherwise = Tuple b a
  
  in x
```

> A workaround is to separate the guard from the constructor pattern:

```purs
f :: Int -> Int -> Int
f a b =
  let 
    tup
      | a < b = Tuple a b
      | otherwise = Tuple b a

    Tuple x y = tup
  
  in x
```

Reply:
Other than this I think it's sensible to merge this while the restriction does exist.